src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/SplashScreen.cs

```diff
@@ -274,160 +275,156 @@ private void Fadeout_Tick(object unused, EventArgs args)
 
-                // initialize the bitmap header
-                MS.Win32.NativeMethods.BITMAPINFO bmInfo = new MS.Win32.NativeMethods.BITMAPINFO(width, height, 32 /*bpp*/);
-                bmInfo.bmiHeader_biCompression = MS.Win32.NativeMethods.BI_RGB;
-                bmInfo.bmiHeader_biSizeImage = (int)(stride * height);
+                uint stride = width * 4;
+
+                // Initialize the bitmap header
+                BITMAPINFOHEADER bmInfo = new()
+                {
+                    biSize = (uint)sizeof(BITMAPINFOHEADER),
+                    biWidth = (int)width,
+                    biHeight = (int)height,
+                    biBitCount = 32,
+                    biPlanes = 1,
+                    biCompression = (uint)BI_COMPRESSION.BI_RGB,
+                    biSizeImage = stride * height
+                };
 
                 // Create a 32bpp DIB.  This DIB must have an alpha channel for UpdateLayeredWindow to succeed.
-                IntPtr pBitmapBits = IntPtr.Zero;
-                _hBitmap = UnsafeNativeMethods.CreateDIBSection(new HandleRef(), ref bmInfo, 0 /* DIB_RGB_COLORS*/, ref pBitmapBits, null, 0);
+                void* pBitmapBits = null;
+
+                _hBitmap = PInvokeCore.CreateDIBSection(default, (BITMAPINFO*)&bmInfo, DIB_USAGE.DIB_RGB_COLORS, &pBitmapBits, HANDLE.Null, 0);
+
+                if (_hBitmap.IsNull)
+                {
+                    throw new Win32Exception();
+                }
 
                 // Copy the decoded image to the new buffer which backs the HBITMAP
-                Int32Rect rect = new Int32Rect(0, 0, width, height);
-                UnsafeNativeMethods.HRESULT.Check(
-                    UnsafeNativeMethods.WIC.CopyPixels(pBitmapFlipRotator, ref rect, stride, stride * height, pBitmapBits));
+                WICRect rect = new()
+                {
+                    X = 0,
+                    Y = 0,
+                    Width = (int)width,
+                    Height = (int)height
+                };
 
-                _hwnd = CreateWindow(_hBitmap, width, height, topMost);
+                pBitmapFlipRotator.Value->CopyPixels(&rect, stride, stride * height, (byte*)pBitmapBits).ThrowOnFailure();
 
-                bSuccess = true;
+                _hwnd = CreateWindow(_hBitmap, (int)width, (int)height, topMost);
+
+                success = true;
             }
             finally
             {
-                if (pImagingFactory != IntPtr.Zero)
-                {
-                    Marshal.Release(pImagingFactory);
-                }
-                if (pDecoder != IntPtr.Zero)
-                {
-                    Marshal.Release(pDecoder);
-                }
-                if (pIStream != IntPtr.Zero)
-                {
-                    Marshal.Release(pIStream);
-                }
-                if (pDecodedFrame != IntPtr.Zero)
+                if (!success)
```

Comment:
Should this be a catch? It looks like the only time success = false is if an exception occurs.

Reply:
Yeah, that's simpler.

Reply:
And this never would return anything but `true` or throw.